internal/cmd/test.go

```diff
@@ -79,7 +87,13 @@ func RunCMDTest(test *testing.T, mainFunc func(), args []string, logLevel log.Lo
 	exit.SetShouldExitForTesting(false)
 	defer exit.SetShouldExitForTesting(true)
 
+	// Prevent the cmd testing dir from automatic cleanup to use during testing.
+	util.SetShouldRemoveTempDirs(false)
+	defer util.SetShouldRemoveTempDirs(true)
+
 	tempDir := util.MustMkDirTemp("autograder-testing-cmd-")
+	defer util.RemoveRecordedTempDirs()
```

Comment:
! Bug

First, before the bug, this line should be grouped with 91/92.
All these are about manipulating how temp dirs are removed during this test.
94 is just a normal line that creates a temp dir using the standard procedure,
it should not be grouped into the "doing unusual things with temp dirs" group.

Now the bug.
Remember that defers are kept as a stack, so later ones happen first.
This will run before the defer on 92.